### PR TITLE
feat: implement button component

### PR DESCRIPTION
**Pull request by bolbolamir**
abounding-position.surge.sh

### CODE REVIEW

File: src/base.css [hunk 1]
```diff
@@ -0,0 +1,56 @@
+@import "modern-normalize/modern-normalize.css";
+
+@font-face {
```
Comment (hasbeenwannabe) [inline src/base.css:3]: Also set `font-display` to `swap` or `optional`.

File: src/components/button.css [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  border-style: solid;
+  border-radius: var(--normal-button-border-radius);
+  border-width: var(--button-border-width);
+  padding-block-start: var(--button-block-padding);
+  padding-inline-end: var(--button-inline-padding);
+  padding-block-end: var(--button-block-padding);
+  padding-inline-start: var(--button-inline-padding);
```
Comment (hasbeenwannabe) [inline src/components/button.css:43]: These should all come from your global design tokens. Same with every other color below.

File: src/components/button.css [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  --faded-button-active-color: hsla(210, 14%, 11%, 0.2);
+
+  --accent-button-color: hsl(321deg 100% 50%);
+  --accent-button-text-color: hsl(0, 0%, 100%);
+  --accent-button-hover-color: hsl(325deg 100% 61%);
+  --accent-button-active-color: hsl(320, 100%, 44%);
+
+  --primary-button-color: hsl(0, 0%, 100%);
+  --primary-button-text-color: hsl(220deg 5% 34%);
+  --primary-button-hover-color: hsl(240, 33%, 99%);
```
Comment (hasbeenwannabe) [inline src/components/button.css:21]: These colors should come from a global color palette.

File: src/components/button.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { Button } from "./button"
+import { OrderIcon, LeftPinkFlashIcon, UserIconFaded } from "./svg"
+
+export const WithDefaultVariant = () => <Button>پیتزا</Button>
+
+export const WithFadedVariantRTL = () => <Button variant="faded">انصراف</Button>
```
Comment (hasbeenwannabe) [inline src/components/button.stories.tsx]: Try to keep story scopes to a single property, e.g. one for each `variant` and one for each `dir`.

File: src/components/button.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+export const WithAccentVariantConstraintedIconRTL = () => (
+  <div style={{ inlineSize: "280px" }}>
+    <Button dir="rtl" icon={<UserIconFaded />} variant="accent">
+      ورود به حساب کاربری
+    </Button>
+  </div>
+)
+
+export const WithAccentRuondedIconRTL = () => (
```
Comment (hasbeenwannabe) [inline src/components/button.stories.tsx]: Typo: Rounded

File: src/components/button.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  rounding = "normal",
+  variant = "faded",
+  children,
+  className: extraClassName,
+  ...otherProps
+}: ButtonProps) => {
+  return (
+    <button
+      className={
+        !extraClassName ? "button" : "button" + " " + `${extraClassName}`
```
Comment (hasbeenwannabe) [inline src/components/button.tsx]: The template literals do nothing here. You could just use `button + " " + extraClassName`, or preferably `'button ${extraClassName}'`. Also suggest not renaming `className`.

File: src/components/button.css [hunk 7]
```diff
@@ -0,0 +1,113 @@
+@import "../base.css";
```
Comment (hasbeenwannabe) [inline src/components/button.css]: This shouldn't be imported here. See [https://ladle.dev/docs/css](https://ladle.dev/docs/css).

File: src/base.css (hunk 1 above)
Comment (bolbolamir) [reply src/base.css:3]: done

File: src/components/button.css (hunk 3 above)
Comment (bolbolamir) [reply src/components/button.css:21]: will fix them in the upcoming design-token branch

File: src/components/button.stories.tsx (hunk 5 above)
Comment (bolbolamir) [reply src/components/button.stories.tsx]: done

File: src/components/button.css (hunk 7 above)
Comment (bolbolamir) [reply src/components/button.css]: done

File: src/components/button.css (hunk 2 above)
Comment (bolbolamir) [reply src/components/button.css:43]: same as above

**Pull request merged**